src/org/labkey/remoteapi/assay/Protocol.java

```diff
@@ -0,0 +1,415 @@
+            result.put("editableRuns", _editableRuns);
+        if (_autoCopyTargetContainerId != null)
+            result.put("autoCopyTargetContainerId", _autoCopyTargetContainerId);
+
+        if (_availableDetectionMethods != null)
+        {
+            JSONArray detectionMethods = new JSONArray();
+            result.put("availableDetectionMethods", detectionMethods);
+            for (String detectionMethod : _availableDetectionMethods)
+                detectionMethods.add(detectionMethod);
+        }
+        if (_selectedDetectionMethod != null)
+            result.put("selectedDetectionMethod", _selectedDetectionMethod);
+
+        if (_availableMetadataInputFormats != null)
+            result.put("availableMetadataInputFormats", _availableMetadataInputFormats);
+        if (_selectedMetadataInputFormat != null)
+            result.put("selectedMetadataInputFormat", _selectedMetadataInputFormat);
+
+        if (_availablePlateTemplates != null)
+        {
+            JSONArray plateTemplates = new JSONArray();
+            result.put("availablePlateTemplates", plateTemplates);
+            for (String plateTemplate : _availablePlateTemplates)
+                plateTemplates.add(plateTemplate);
+        }
+        if (_selectedPlateTemplate != null)
+            result.put("seletedPlateTemplate", _selectedPlateTemplate);
+
+        if (_protocolParameters != null)
+            result.put("protocolParameters", _protocolParameters);
+
+        if (_protocolTransformScripts != null)
+        {
+            JSONArray transformScripts = new JSONArray();
+            result.put("protocolTransformScripts", transformScripts);
+            for (String script : _protocolTransformScripts)
+                transformScripts.add(script);
+        }
+
+        return result;
+    }
+
+    public Long getProtocolId()
+    {
+        return _protocolId;
+    }
+
+    public String getName()
+    {
+        return _name;
+    }
+
+    public Protocol setName(String name)
+    {
+        _name = name;
+        return this;
+    }
+
+    public String getDescription()
+    {
+        return _description;
+    }
+
+    public void setDescription(String description)
+    {
+        _description = description;
+    }
+
+    public String getProviderName()
+    {
+        return _providerName;
+    }
+
+    public void setProviderName(String providerName)
+    {
+        _providerName = providerName;
+    }
+
+    public List<Domain> getDomains()
+    {
+        return _domains;
+    }
+
+    public void setDomains(List<Domain> domains)
+    {
+        _domains = domains;
+    }
+
+    public Protocol setAllowBackgroundUpload(boolean allowBackgroundUpload)
+    {
+        _allowBackgroundUpload = allowBackgroundUpload;
+        return this;
+    }
+    public Boolean getAllowBackgroundUpload()
+    {
+        return _allowBackgroundUpload;
+    }
+
+    public Protocol setAllowEditableResults(boolean allowEditableResults)
```

Comment:
Similar to the comments above, if some of these properties are not editable, we should probably remove the setters for them.